packages/api/src/command/medical/patient/consolidated-delete.ts

```diff
@@ -0,0 +1,112 @@
+import { OperationOutcomeError } from "@medplum/core";
+import { BundleEntry, OperationOutcomeIssue, Resource } from "@medplum/fhirtypes";
+import { ResourceTypeForConsolidation } from "@metriport/api-sdk";
+import { executeAsynchronously } from "@metriport/core/util/concurrency";
+import { Patient } from "../../../domain/medical/patient";
+import { makeFhirApi } from "../../../external/fhir/api/api-factory";
+import { isResourceDerivedFromDocRef } from "../../../external/fhir/shared";
+import { capture } from "../../../shared/notifications";
+import { Util } from "../../../shared/util";
+import { getConsolidated } from "./consolidated-get";
+
+const numberOfParallelExecutions = 10;
+
+export type DeleteConsolidatedFilters = {
+  resources?: ResourceTypeForConsolidation[];
+  docIds: string[];
+};
+
+export type DeleteConsolidatedParams = {
+  patient: Pick<Patient, "id" | "cxId" | "data">;
+} & DeleteConsolidatedFilters;
+
+export async function deleteConsolidated(params: DeleteConsolidatedParams): Promise<void> {
+  const { patient, resources, docIds } = params;
+  const { log } = Util.out(`deleteConsolidated - cxId ${patient.cxId}, patientId ${patient.id}`);
+  const fhir = makeFhirApi(patient.cxId);
+
+  const getResourcesToDelete = async () => {
+    try {
+      const bundle = await getConsolidated({
+        patient,
+        resources,
+      });
+      const resourcesFromFHIRServer = bundle.bundle.entry;
+      if (!resourcesFromFHIRServer || resourcesFromFHIRServer.length <= 0) {
+        log(`No resources to delete`);
+        return;
+      }
+
+      const isDerivedFromDocRefs = (r: Resource) =>
+        docIds.some(id => isResourceDerivedFromDocRef(r, id));
+
+      const resourcesToDelete = resourcesFromFHIRServer.filter(r => {
+        const resource = r.resource;
+        if (!resource) return false;
+        return resource.resourceType !== "DocumentReference" && isDerivedFromDocRefs(resource);
+      });
+      return resourcesToDelete;
+    } catch (error) {
+      const msg = `Failed to get consolidated FHIR resources ;`;
```

Comment:
nit: the `;` will result in weird formatting here, ie `;:`

Reply:
good catch, it's a typo